In [0]:
# Import Tensorflow
import tensorflow as tf

In [0]:
# Load & prepare MNIST dataset
mnist = tf.keras.datasets.mnist

(x_train, y_train), (x_test, y_test) = mnist.load_data()
x_train, x_test = x_train / 255.0, x_test / 255.0

In [0]:
# Build a model by stacking layers
model = tf.keras.models.Sequential([
  tf.keras.layers.Flatten(input_shape=(28, 28)),
  tf.keras.layers.Dense(128, activation='relu'),
  tf.keras.layers.Dropout(0.2),
  tf.keras.layers.Dense(10)
])

In [0]:
predictions = model(x_train[:1]).numpy()
predictions

array([[-0.11829307,  0.24285328,  1.0221195 ,  0.59569883,  0.45171282,
         0.2540081 , -0.2036432 ,  0.21878558,  0.26732546,  0.32637215]],
      dtype=float32)

In [0]:
# Take vector of logits and return scalar loss
loss_fn = tf.keras.losses.SparseCategoricalCrossentropy(from_logits=True)

In [0]:
model.compile(optimizer='adam',
              loss=loss_fn,
              metrics=['accuracy'])

In [0]:
model.fit(x_train, y_train, epochs=5)

Epoch 1/5
1875/1875 [==============================] - 4s 2ms/step - loss: 0.2969 - accuracy: 0.9137
Epoch 2/5
1875/1875 [==============================] - 4s 2ms/step - loss: 0.1439 - accuracy: 0.9561
Epoch 3/5
1875/1875 [==============================] - 5s 3ms/step - loss: 0.1078 - accuracy: 0.9668
Epoch 4/5
1875/1875 [==============================] - 5s 2ms/step - loss: 0.0888 - accuracy: 0.9722
Epoch 5/5
1875/1875 [==============================] - 4s 2ms/step - loss: 0.0756 - accuracy: 0.9759


In [0]:
model.evaluate(x_test,  y_test, verbose=2)

313/313 - 0s - loss: 0.0760 - accuracy: 0.9773


[0.07599371671676636, 0.9772999882698059]

The model's accuracy is appx 98%, which is a great score.